src/macros.jl

```diff
@@ -2748,3 +2748,192 @@ julia> @rdistinct!(df, :x + :y)
 macro rdistinct!(d, args...)
     esc(rdistinct!_helper(d, args...))
 end
+
+##############################################################################
+##
+## @rename - rename columns with keyword args 
+##
+##############################################################################
+
+function pair_to_string(p...)        
+    str_p = Vector{Pair{String, String}}(undef,length(p))
+    for (i,j) in enumerate(p)                    
+        str_p[i] = string(j[1])=> string(j[2])     
+    end    
+    return str_p
+end
+
+function make_renamed(x, @nospecialize(t...))   
+    p = pair_to_string(t...)     
+    DataFrames.rename(x,p...)
+end
+
+function rename_helper(x, args...)
+    x, exprs, outer_flags, kw = get_df_args_kwargs(x, args...; wrap_byrow = false)    
+    t = (fun_to_vec(ex; gensym_names = false,  outer_flags=outer_flags) for ex in exprs)
+    quote                  
+        $make_renamed($x,$(t...))
+    end
+end
+
+"""
+    @rename(d, args...)
+
+Change column names.
+
+### Arguments
+
+* `d` : an AbstractDataFrame
+* `args...` : expressions of the form `:new = :old` specifying the change of a column's name 
+from "old" to "new"
+
+### Returns
+
+* `::AbstractDataFrame`
+
+Inputs to `@rename` can come in two formats: a `begin ... end` block, or as a series of
+arguments and keyword-like arguments. For example, the following are equivalent:
```

Comment:
it is not clear for me in the docstring what "arguments" means and what "keyword-like arguments" means. In particular if we have a form `lhs = rhs` what values are accepted on both sides.

Reply:
it would be also good too mention that whole expressions can be escaped by `$` (like in the tests).

Reply:
Hi, @bkamins. Thanks for the review! I've updated the docs to clarify phrasing regarding the points you raised above. I'm happy to continue revising as you see fit. Thanks!

Reply:
Looks good. Thank you!